src/models.ts

```diff
@@ -4,6 +4,23 @@ export type PokemonUri = {
   url: string;
 };
 
+export type KnownPokemonType =
+  | "bug"
+  | "electric"
+  | "fairy"
+  | "fighting"
+  | "fire"
+  | "flying"
+  | "grass"
+  | "ground"
+  | "ice"
+  | "normal"
+  | "poison"
+  | "psychic"
+  | "rock"
+  | "steel"
+  | "water";
```

Comment:
TypeScript tip: It's incredibly useful to define specific types of strings rather than just any string ... like "ok, its a string, but not **any** string but one of these" 

you'll find this pattern is super valuable in Ts